### DISCUSSION TITLE
v3.1.5

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#1020] [Announcements]
## What's Changed
* Adding a ranking step of research sources by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/992
* UX Fixes by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/994
* Fix max_results for Google Search by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1009
* Add xAI by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1017
* Fix uncaught exception in Google retrieval by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1018
* Default to the retriever set in config rather than just tavily by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1019
* 【feature】Commit Message: Optimized PyMuPDFScraper to handle invalid o… by @MC-shark in https://github.com/assafelovic/gpt-researcher/pull/1012
* Add support for DashScope provider in LLM and embeddings modules by @AStupidBear in https://github.com/assafelovic/gpt-researcher/pull/1013
* Fix AWS Bedrock invocation; create output folder in cli.py if it doesn't exist by @ai-1st in https://github.com/assafelovic/gpt-researcher/pull/1014

## New Contributors
* @winsonluk made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/1009
* @MC-shark made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/1012
* @AStupidBear made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/1013
* @ai-1st made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/1014

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v3.1.4...v3.1.5

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v3.1.5'>v3.1.5</a>.</em>

**Comment by EmminiX** [comment]
Can't thank you enough for this amazing tool! You are amazing!
